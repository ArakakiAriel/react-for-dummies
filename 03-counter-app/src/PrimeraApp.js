//Nuestra primera functional component
//Generalmente esperamos que devuelva algo de HTML para renderizar en nuestra webpage
import React from 'react';
import PropTypes from 'prop-types';
//import React, { Fragment } from 'react'; //Siempre que utilizemos html en nuestro archivo js debemos importar react
//const PrimeraApp = ( {saludo = 'Hola Mundo'} ) => { //Utilizamos desestructuración para poder utilizar directamente el nombre de nuestra propiedad
                        // => = 'Hola Mundo' es el valor por defecto de la prop "saludo" por si no le llega ningun valor.
    //const saludo = 'Hola Mundo';

const PrimeraApp = ( {saludo, subtitulo} ) => {

    console.log(saludo)

    // const objeto = {
    //     nombre: 'Kenji',
    //     edad: 27
    // };

    return ( //Utilizar los paréntesis para poder utilizar más de una linea para retornar
        //<Fragment> 
        <>
            <h1>{saludo}</h1>
            {/* <pre>{JSON.stringify(objeto, null, 3)}</pre>  */}
            <p>{subtitulo}</p>
        </>
        //</Fragment>
    ); // En caso de querer retornar más de un elemento utilizaremos el Fragment para encapsularlos

}

PrimeraApp.propTypes = {
     saludo: PropTypes.string.isRequired,
     subtitulo: PropTypes.string
}
PrimeraApp.defaultProps = { //En caso de no mandarse alguna propiedad, acá se le puede definir el default
    subtitulo: 'Proyecto básico de React.'
}

export default PrimeraApp;