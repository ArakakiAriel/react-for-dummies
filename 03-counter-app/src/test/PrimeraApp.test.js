import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme"; //Importamos shallow
import React from "react";
describe("PrimeraApp", () => {
  test("debe mostrar <PrimeraApp /> correctamente", () => {
    const saludo = "Hola, soy Kenji";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot(); //u
  });
  test("debe mostrar el subtitulo enviado por props", () => {
    const saludo = "Hola, soy Kenji";
    const subtitulo = "Proyecto básico de React.";
    const wrapper = shallow(
        <PrimeraApp 
            saludo={saludo}
            subtitulo={subtitulo}
        />
    );
    
    const textoParrafo = wrapper.find('p').text(); //con el find podemos buscar por elemento, nos va a devolver un array de elementos en caso de encontrarse con más de uno
    expect(textoParrafo).toBe(subtitulo);


    expect(wrapper).toMatchSnapshot(); //
  });
});
