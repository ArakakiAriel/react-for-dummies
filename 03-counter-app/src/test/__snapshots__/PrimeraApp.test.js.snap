// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeraApp debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Kenji
  </h1>
  <p>
    Proyecto básico de React.
  </p>
</Fragment>
`;

exports[`PrimeraApp debe mostrar el subtitulo enviado por props 1`] = `
<Fragment>
  <h1>
    Hola, soy Kenji
  </h1>
  <p>
    Proyecto básico de React.
  </p>
</Fragment>
`;
